File: app/src/main/java/com/adityagupta/trekki/Accelerometer.java
package com.adityagupta.trekki;

public class Accelerometer {
    public float X;
    public float Y;
    public float Z;
    public double R;


    public Accelerometer(float[] event) {
        X = event[0];
        Y = event[1];
        Z = event[2];
        R = Math.sqrt(X*X + Y*Y + Z*Z);
    }

    public Number toNumber() {
        Number number = R;
        return number;
    }
}

File: app/src/main/java/com/adityagupta/trekki/Accelerometer.kt
package com.adityagupta.trekki

import kotlin.math.sqrt

class Accelerometer(event: FloatArray?) {
    var X: Float = event?.get(0) ?: 0f
    var Y: Float = event?.get(1) ?: 0f
    var Z: Float = event?.get(2) ?: 0f
    var R: Double

    fun toNumber(): Number {
        return R
    }

    init {
        R = sqrt(X * X + Y * Y + (Z * Z).toDouble())
    }
}